packages/fluentui/react-northstar/src/components/RadioGroup/RadioGroupItem.tsx

```diff
@@ -76,93 +84,124 @@ export const radioGroupItemSlotClassNames: RadioGroupItemSlotClassNames = {
   indicator: `${radioGroupItemClassName}__indicator`,
 };
 
-class RadioGroupItem extends AutoControlledComponent<WithAsProp<RadioGroupItemProps>, RadioGroupItemState> {
-  elementRef = React.createRef<HTMLElement>();
-
-  static create: ShorthandFactory<RadioGroupItemProps>;
-
-  static displayName = 'RadioGroupItem';
-
-  static deprecated_className = radioGroupItemClassName;
-
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false,
-    }),
-    checked: PropTypes.bool,
-    defaultChecked: PropTypes.bool,
-    disabled: PropTypes.bool,
-    indicator: customPropTypes.shorthandAllowingChildren,
-    label: customPropTypes.itemShorthand,
-    name: PropTypes.string,
-    onClick: PropTypes.func,
-    onChange: PropTypes.func,
-    shouldFocus: PropTypes.bool,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    vertical: PropTypes.bool,
-  };
-
-  static defaultProps = {
-    accessibility: radioGroupItemBehavior,
-    indicator: {},
+export type RadioGroupItemStylesProps = Required<Pick<RadioGroupItemProps, 'disabled' | 'vertical' | 'checked'>>;
+
+const RadioGroupItem: React.FC<WithAsProp<RadioGroupItemProps>> &
+  FluentComponentStaticProps<RadioGroupItemProps> = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(RadioGroupItem.displayName, context.telemetry);
+  setStart();
+  const { label, indicator, disabled, vertical, className, design, styles, variables } = props;
+  const elementRef = React.useRef<HTMLElement>();
+  const ElementType = getElementType(props);
+  const unhandledProps = useUnhandledProps(RadioGroupItem.handledProps, props);
+
+  const [checked, setChecked] = useAutoControlled({
+    defaultValue: props.defaultChecked,
+    value: props.checked,
+    initialValue: false,
+  });
+
+  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+    _.invoke(props, 'onClick', e, props);
+    setChecked(prevChecked => !prevChecked);
   };
 
-  static autoControlledProps = ['checked'];
-
-  actionHandlers = {
-    performClick: e => {
-      e.preventDefault();
-      this.handleClick(e);
+  React.useEffect(() => {
+    checked && props.shouldFocus && elementRef.current.focus();
+    _.invoke(props, 'onChange', undefined, { ...props, checked });
```

Comment:
I would rather move this to `handleClick` as this looks pretty strange and unexpected

Reply:
It was my first thought as well, but then it doesn't work... The focus never change.

Reply:
Can you please clarify what you have tried there? What change you applied?

Reply:
Tried to move the condition and `.focus()` inside the click to either the event target and to the element ref. Neither applied the focus change when navigating the RadioGroupItem

Reply:
This is what we get when trying to move the `focus()` to inside `handleClick`

<img width="445" alt="Screenshot 2020-05-25 at 11 00 24" src="https://user-images.githubusercontent.com/8545105/82797566-8baeb180-9e77-11ea-9ac4-90318bd68870.png">

Reply:
Have you tried these changes?

```diff
  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
    _.invoke(props, 'onClick', e, props);
-    setChecked(prevChecked => !prevChecked);
+   setChecked(prevChecked => {
+        _.invoke(props, 'onChange', undefined, { ...props, checked: !prevChecked })
+        return !prevChecked
+      });
//
    if (checked && shouldFocus) elementRef.current.focus();
-   _.invoke(props, 'onChange', undefined, { ...props, checked });
```

Reply:
I thought you meant to move `if (checked && shouldFocus) elementRef.current.focus();` inside the click.

Reply:
This diff works.